android/src/com/mapswithme/maps/editor/ViralFragment.java

```diff
@@ -0,0 +1,98 @@
+package com.mapswithme.maps.editor;
+
+import android.annotation.SuppressLint;
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
+import android.support.v7.app.AlertDialog;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.TextView;
+
+import java.util.Random;
+
+import com.mapswithme.maps.MwmApplication;
+import com.mapswithme.maps.R;
+import com.mapswithme.maps.base.BaseMwmDialogFragment;
+import com.mapswithme.util.ConnectionState;
+
+public class ViralFragment extends BaseMwmDialogFragment
+{
+  private static final String EXTRA_CONTRATS_SHOWN = "CongratsShown";
+  private TextView mViral;
+
+  @StringRes
+  private static int sViralText;
+
+  public static boolean shouldDisplay()
+  {
+    return !MwmApplication.prefs().contains(EXTRA_CONTRATS_SHOWN) &&
+           Editor.nativeGetStats()[0] >= 2 &&
+           ConnectionState.isConnected();
+  }
+
+  @NonNull
+  @Override
+  public Dialog onCreateDialog(Bundle savedInstanceState)
+  {
+    MwmApplication.prefs().edit().putBoolean(EXTRA_CONTRATS_SHOWN, true).apply();
+
+    final AlertDialog.Builder builder = new AlertDialog.Builder(getActivity()).setCancelable(true);
+    final LayoutInflater inflater = LayoutInflater.from(getActivity());
+    @SuppressLint("InflateParams")
+    final View root = inflater.inflate(R.layout.fragment_editor_viral, null);
+    mViral = (TextView) root.findViewById(R.id.viral);
+    initViralText();
+    mViral.setText(sViralText);
+    // TODO set rank with correct text.
+    root.findViewById(R.id.tell_friend).setOnClickListener(new View.OnClickListener()
+    {
+      @Override
+      public void onClick(View v)
+      {
+        dismiss();
+        // TODO send statistics
+        // TODO open some share link
+      }
+    });
+    root.findViewById(R.id.close).setOnClickListener(new View.OnClickListener()
+    {
+      @Override
+      public void onClick(View v)
+      {
+        dismiss();
+        // TODO send statistics
+      }
+    });
+    return builder.setView(root).create();
+  }
+
+  @Override
+  public void onDismiss(DialogInterface dialog)
+  {
+    super.onDismiss(dialog);
+    // TODO remove before merge
+  }
+
+  private void initViralText()
+  {
+    if (sViralText != 0)
+      return;
+
+    if (new Random().nextBoolean())
+      // FIXME
+//        sViralText = R.string.editor_done_dialog_1;
+      sViralText = R.string.dialog_routing_change_start;
+    else
+//        sViralText = R.string.editor_done_dialog_2;
+      sViralText = R.string.dialog_routing_change_end;
+  }
+
+  // Counts fake editor rank based on number of total edits made by user.
```

Comment:
Тут можно подробней почитать https://confluence.mail.ru/display/MAPSME/Meta+Info+Editor+v1#MetaInfoEditorv1-Socializing <https://confluence.mail.ru/display/MAPSME/Meta+Info+Editor+v1#MetaInfoEditorv1-Socializing>
> On Mar 18, 2016, at 5:52 PM, Dmitry Yunitsky <dmitry.yunitsky@gmail.com> wrote:
> 
> Ну вот так вот) Считаем некий виртуальный рейтинг пользователя, основываясь на числе его правок в осм) 
>> On Mar 18, 2016, at 5:51 PM, Alexander Marchuk <notifications@github.com <mailto:notifications@github.com>> wrote:
>> 
>> In android/src/com/mapswithme/maps/editor/ViralFragment.java <https://github.com/mapsme/omim/pull/2338#discussion_r56667147>:
>> 
>> > +
>> > +  private void initViralText()
>> > +  {
>> > +    if (sViralText != 0)
>> > +      return;
>> > +
>> > +    if (new Random().nextBoolean())
>> > +      // FIXME
>> > +//        sViralText = R.string.editor_done_dialog_1;
>> > +      sViralText = R.string.dialog_routing_change_start;
>> > +    else
>> > +//        sViralText = R.string.editor_done_dialog_2;
>> > +      sViralText = R.string.dialog_routing_change_end;
>> > +  }
>> > +
>> > +  // Counts fake editor rank based on number of total edits made by user.
>> Шта?
>> 
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2338/files/49a98513afd524db41d893d7cec128cfb271218e..b0dcfabe82c6bad90154275d1b20f2651bbdf1b1#r56667147>
>